primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -292,26 +303,31 @@ private static boolean isValidFileContent(PrimeApplicationContext context, FileU
             }
         }
         finally {
-            try {
-                Files.delete(tempFile);
-            }
-            catch (Exception ex) {
-                if (LOGGER.isLoggable(Level.WARNING)) {
-                    LOGGER.log(Level.WARNING, String.format("Could not delete temporary file %s, will try to delete on JVM exit",
-                            tempFile.toAbsolutePath()), ex);
-                    try {
-                        tempFile.toFile().deleteOnExit();
-                    }
-                    catch (Exception ex1) {
-                        if (LOGGER.isLoggable(Level.WARNING)) {
-                            LOGGER.log(Level.WARNING, String.format("Could not register temporary file %s for deletion on JVM exit",
-                                    tempFile.toAbsolutePath()), ex1);
-                        }
+            deleteTempFile(tempFile);
+            deleteTempFile(tempDirectory);
```

Comment:
if tempfile in tempdirectory, deleting the directory should be enough?

Reply:
I delete the file separately first because:

>If the file is a directory then the directory must be empty.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#delete(java.nio.file.Path)